Framework/Kernel/src/PropertyManager.cpp

```diff
@@ -36,6 +37,32 @@ const std::string createKey(const std::string &name) {
 }
 } // namespace
 
+const std::string PropertyManager::INVALID_VALUES_SUFFIX = "_invalid_values";
+/// Gets the correct log name for the matching invalid values log for a given
+/// log name
+std::string
+PropertyManager::getInvalidValuesFilterLogName(const std::string &logName) {
+  return logName + PropertyManager::INVALID_VALUES_SUFFIX;
+}
+std::string
+PropertyManager::getLogNameFromInvalidValuesFilter(const std::string &logName) {
+  std::string retVal = "";
+  if (PropertyManager::isAnInvalidValuesFilterLog(logName)) {
+    retVal = logName.substr(
+        0, logName.size() - PropertyManager::INVALID_VALUES_SUFFIX.size());
+  }
+  return retVal;
+}
+bool PropertyManager::isAnInvalidValuesFilterLog(const std::string &logName) {
+  const auto fullString = logName;
```

Comment:
I hadn't spotted this before. Can you help me understand why we take this copy?

Reply:
Hmm, no good reason, I'll remove it.

Reply:
Thanks.